Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs

```diff
@@ -73,48 +91,150 @@ private void PullingWire()
+        {
+            plugAnchor.transform.position = gunController.Player.transform.position;
+            plugAnchor.transform.forward = hit.normal;
+            rope.enabled = true;
+            canThrow = false;
+            LeanTween.value(gameObject, SetThrowValue, 0f, ropeLength, 0.02f * ropeLength);
+            plugAnchor.gameObject.LeanMove(hit.point, 0.02f * hit.distance)
+                .setOnComplete(() =>
+                {
+                    rope.ResetRope(plugAnchor.WireOrigin);
+                    isThrowing = false;
+                    isWired = true;
+                    plugAnchor.SetPulseStrength(0.85f);
+                    gunController.stats.Ammo = gunController.stats.MagazineSize;
+                });
+            rope.ResetRope(plugAnchor.WireOrigin);
+            plugAnchor.gameObject.SetActive(true);
+        }
+        else
+        {
+            plugAnchor.SetPulseStrength(0);
+            plugAnchor.transform.position = gunController.Player.transform.position;
+            plugAnchor.transform.forward = gunController.Player.inputManager.transform.forward;
+            rope.enabled = true;
+            plugAnchor.gameObject.SetActive(true);
+            AnimateFullPullBack();
+        }
+    }
+
+    public void PlayThrowAudio()
+    {
+        audioSource.clip = plugThrow;
+        audioSource.Play();
+    }
+
+    private void AnimateFullPullBack()
+    {
+        LeanTween.value(gameObject, SetThrowValue, 0f, ropeLength, 0.02f * ropeLength)
+            .setEaseOutElastic();
+
+        plugAnchor.gameObject.LeanMove(gunController.Player.inputManager.transform.position + gunController.Player.inputManager.transform.forward * ropeLength, 0.02f * ropeLength)
+            .setOnComplete(() =>
+            {
+                retractingWire.Play(audioSource);
+                rope.CollisionCheckActive = false;
+                LeanTween.value(gameObject, SetPullbackValue, 1f, 0f, 1f)
+                .setOnComplete(() =>
+                {
+                    rope.enabled = false;
+                    isThrowing = false;
+                    plugAnchor.gameObject.SetActive(false);
+                    canThrow = true;
+                    rope.CollisionCheckActive = true;
+                });
+            });
+    }
+    private void AnimateLastVertex()
+    {
+        retractingWire.Play(audioSource);
+        rope.CollisionCheckActive = false;
+        LeanTween.value(gameObject, SetPullbackValue, 1f, 0f, 1f)
+            .setOnComplete(() =>
+            {
+                    rope.enabled = false;
+                    isThrowing = false;
+                    plugAnchor.gameObject.SetActive(false);
+                    canThrow = true;
+                    rope.CollisionCheckActive = true;
+            });
+    }
+
+    public void SetThrowValue(float value)
```

Comment:
SetThrowValue? What does this even do? The value is unused and the method name is a lil confusing considering it just resets the rope 🤔

Reply:
Renamed